account_bank_statement_import_adyen/models/account_bank_statement_import.py

```diff
@@ -43,6 +43,23 @@ def balance(self, row):
             row[i] if row[i] else 0.0
             for i in (16, 17, 18, 19, 20))
 
+    @api.model
+    def _import_adyen_transaction_find_partner(
+            self, statement, statement_id, row
+    ):
+        """ Return exactly one partner matched for this transaction, or an
+        empty recordset """
+
+        return self.env['account.move.line'].search([
+            ('account_id.reconcile', '=', True),
+            ('account_id.internal_type', '=', 'receivable'),
+            ('reconciled', '=', False),
+            ('balance', '=', self.balance(row)),
```

Comment:
Is this not impacted by Python's representation of floats? I was expecting to see a float comparison of found lines instead.

Reply:
this got me down a nice rabbit hole, and while I didn't find a definite source saying it's okay to do that, I think I've a good argument: The monetary types are [numeric](https://github.com/OCA/OCB/blob/12.0/odoo/fields.py#L1365), and given all values are rounded before writing, and we parse rounded values, I think we're fine.

Kind of evidence:

```sql
vanmoof_test=> create table test_numeric(a numeric, b float);
CREATE TABLE
vanmoof_test=> insert into test_numeric (a, b) values (1::numeric/3, 1::float/3);
INSERT 0 1
vanmoof_test=> select * from test_numeric;
           a            |         b         
------------------------+-------------------
                      0 |                 0
 0.33333333333333333333 | 0.333333333333333
(2 rows)

vanmoof_test=> select * from test_numeric where a=1::float/3;
           a            |         b         
------------------------+-------------------
 0.33333333333333333333 | 0.333333333333333
(1 row)
```

Reply:
Ok, thanks for the deep dive!